easybuild/easyblocks/generic/dataset.py

```diff
@@ -0,0 +1,100 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for installing datasets
+
+@author: Samuel Moors (Vrije Universiteit Brussel)
+"""
+import os
+
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.easyblocks.generic.binary import Binary
+from easybuild.framework.easyconfig.default import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import compute_checksum, create_index, is_readable, mkdir, move_file, remove_file
+from easybuild.tools.filetools import symlink
+from easybuild.tools.utilities import trace_msg
+
+
+class Dataset(Binary):
+    """Support for installing datasets"""
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Extra easyconfig parameters specific to Data easyblock."""
+        extra_vars = EasyBlock.extra_options(extra_vars)
+        extra_vars.update({
+            'extract_sources': [True, "Whether or not to extract data sources", CUSTOM],
+            'data_install_path': [None, "Custom installation path for datasets", CUSTOM],
+            'cleanup_data_sources': [False, "Whether or not to delete the data sources after installation", CUSTOM]
+        })
+        return extra_vars
+
+    def __init__(self, *args, **kwargs):
+        """Initialize Dataset-specific variables."""
+        super().__init__(*args, **kwargs)
+
+        if self.cfg['sources']:
+            raise EasyBuildError(
+                "Easyconfig parameter 'sources' is not supported for this EasyBlock. Use 'data_sources' instead.")
+
+        if self.cfg['data_install_path']:
+            self.installdir = self.cfg['data_install_path']
+
+        # extract/copy sources directly into installation directory
+        self.build_in_installdir = True
+
+    def install_step(self):
+        """No install step, datasets are extracted directly into installdir"""
+        pass
+
+    def post_processing_step(self):
+        """Add files to object_storage, remove duplicates, add symlinks"""
+        trace_msg('adding files to object_storage...')
+
+        # creating object storage at root of software name to reuse identical files in different versions
+        object_storage = os.path.join(os.pardir, 'object_storage')
+        datafiles = create_index(os.curdir)
+
+        for datafile in datafiles:
+            cks = compute_checksum(datafile, checksum_type='sha256')
+            objstor_file = os.path.join(
+                object_storage, cks[0], cks[1], cks[2], cks[3], cks[4], cks[5], cks[6], cks[7], cks[8:])
```

Comment:
This may be a bit easier on the eye:
objstor_file = os.path.join(object_storage, os.sep.join(list(cks[:8])), cks[8:])
or
objstor_file = os.path.join(object_storage, os.sep.join(list(cks[:8])), cks)
to get the full cks as the actual filename (as puppet does)